recordings-downloader/csharp/Recordings/Program.cs

```diff
@@ -0,0 +1,193 @@
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Reflection;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Diagnostics;
+using PureCloudPlatform.Client.V2.Api;
+using PureCloudPlatform.Client.V2.Client;
+using PureCloudPlatform.Client.V2.Extensions;
+using PureCloudPlatform.Client.V2.Extensions.Notifications;
+using PureCloudPlatform.Client.V2.Model;
+using Newtonsoft.Json;
+
+namespace Recordings
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            // OAuth input
+            Console.Write("Enter Client ID: ");
+            string clientId = Console.ReadLine();
+            Console.Write("Enter Client Secret: ");
+            string clientSecret = Console.ReadLine();
+
+            // Get the Date Interval
+            Console.Write("Enter Date Interval (YYYY-MM-DDThh:mm:ss/YYYY-MM-DDThh:mm:ss): ");
+            string interval = Console.ReadLine();
+
+            Console.WriteLine("Working...");
+
+            // Configure SDK Settings
+            string accessToken = GetToken(clientId, clientSecret);
+            PureCloudPlatform.Client.V2.Client.Configuration.Default.AccessToken = accessToken;
+
+            // Instantiate APIs
+            ConversationsApi conversationsApi = new ConversationsApi();
+            RecordingApi recordingApi = new RecordingApi();
+
+            // Get Conversation IDs and corresponding Media Type within a date interval
+            Dictionary<string, AnalyticsSession.MediaTypeEnum> conversationsMapIDType = GetConversations(interval, conversationsApi);
+
+            // Use the Conversation IDs and request a batch download for the conversations
+            string jobId = RequestBatchDownload(conversationsMapIDType.Keys.ToList(), recordingApi);
+
+            // Get all conversations with recordings
+            List<BatchDownloadJobResult> jobResults = GetJobResults(jobId, recordingApi);
+
+            // Download all the recordings and separate into folders named after the Conversation ID
```

Comment:
I believe this should use the recording ID, or join the conversation ID and recording ID, for the filename. A conversation can have multiple recordings if dual channel recording is enabled, and I believe also if the automated recording was stopped and started.